File: app/src/main/java/com/iug/coursehub/MainViewModel.java
package com.iug.coursehub;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {

    private final MutableLiveData<UiAction> _uiAction = new MutableLiveData<>();
    private final LiveData<UiAction> uiAction = _uiAction;

    public LiveData<UiAction> getUiAction() {
        return uiAction;
    }

    public void navigateToDashboardNavHost() {
        _uiAction.postValue(UiAction.NAVIGATE_TO_DASHBOARD_NAV_HOST);
    }

    public void navigateToUserNavHost() {
        _uiAction.postValue(UiAction.NAVIGATE_TO_USER_NAV_HOST);
    }

    public void navigateUserToAuth() {
        _uiAction.postValue(UiAction.NAVIGATE_USER_TO_AUTH);
    }

    public enum UiAction {
        NAVIGATE_TO_DASHBOARD_NAV_HOST,
        NAVIGATE_TO_USER_NAV_HOST,
        NAVIGATE_USER_TO_AUTH
    }
}

File: app/src/main/java/com/iug/coursehub/MainViewModel.kt
package com.iug.coursehub

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {

    private val _uiAction = MutableLiveData<UiAction>()
    val uiAction: LiveData<UiAction> = _uiAction

    fun navigateToDashboardNavHost() {
        _uiAction.postValue(UiAction.NavigateToDashboardNavHost)
    }

    fun navigateToUserNavHost() {
        _uiAction.postValue(UiAction.NavigateToUserNavHost)
    }

    fun navigateUserToAuth() {
        _uiAction.postValue(UiAction.NavigateUserToAuth)
    }

    sealed interface UiAction {
        data object NavigateToDashboardNavHost : UiAction
        data object NavigateToUserNavHost : UiAction
        data object NavigateUserToAuth : UiAction
    }
}